src/Microsoft.TestPlatform.CrossPlatEngine/Execution/BaseRunTests.cs

```diff
@@ -88,24 +90,27 @@ protected BaseRunTests(string runSettings, TestExecutionContext testExecutionCon
 
         private void SetContext()
         {
-            this.testRunCache = new TestRunCache(testExecutionContext.FrequencyOfRunStatsChangeEvent, testExecutionContext.RunStatsChangeEventTimeout, this.OnCacheHit);
+            this.testRunCache = new TestRunCache(this.testExecutionContext.FrequencyOfRunStatsChangeEvent, testExecutionContext.RunStatsChangeEventTimeout, this.OnCacheHit);
             this.dataCollectionTestCaseEventManager = new DataCollectionTestCaseEventManager();
 
             this.inProcDataCollectionExtensionManager = new InProcDataCollectionExtensionManager(runSettings, testRunCache, this.dataCollectionTestCaseEventManager);
 
-
-
-            // Verify if datacollection is enabled and wrap the testcasehandler around to get the events 
-            if (inProcDataCollectionExtensionManager.IsInProcDataCollectionEnabled)
+            if (DataCollectionTestCaseEventSender.Instance != null)
             {
-                this.testCaseEventsHandler = new TestCaseEventsHandler(this.dataCollectionTestCaseEventManager, this.testCaseEventsHandler);
+                this.outOfProcDataCollectionManager = new ProxyOutOfProcDataCollectionManager(DataCollectionTestCaseEventSender.Instance, this.dataCollectionTestCaseEventManager);
```

Comment:
This code is not testable. `DataCollectionTestCaseEventSender` is hardcoded static.